bcda/auth/plugin/alpha.go

```diff
@@ -280,40 +280,61 @@ func getACOFromDB(acoUUID string) (models.ACO, error) {
 	return aco, err
 }
 
-func GetParamString(params []byte, name string) (string, error) {
+func GetParamString(name string, params ...interface{}) (string, error) {
 	var (
 		j   interface{}
 		err error
 	)
 
-	if err = json.Unmarshal(params, &j); err != nil {
-		return "", err
-	}
-	paramsMap := j.(map[string]interface{})
+	for _, param := range params {
+		if _, ok := param.(string); !ok {
+			return "", errors.New("invalid string value")
+		}
+
+		if err = json.Unmarshal([]byte(param.(string)), &j); err != nil {
+			return "", err
+		}
+
+		paramsMap, ok := j.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		stringForName, ok := paramsMap[name].(string)
+		if !ok {
+			continue
+		}
 
-	stringForName, ok := paramsMap[name].(string)
-	if !ok {
-		return "", errors.New("missing or otherwise invalid string value for " + name)
+		return stringForName, err
 	}
 
-	return stringForName, err
+	return "", errors.New("missing or otherwise invalid string value for " + name)
 }
 
-func GetParamPositiveInt(params []byte, name string) (int, error) {
+func GetParamPositiveInt(name string, params ...interface{}) (int, error) {
 	var (
 		j   interface{}
 		err error
 	)
 
-	if err = json.Unmarshal(params, &j); err != nil {
-		return -1, err
-	}
-	paramsMap := j.(map[string]interface{})
+	for _, param := range params {
+		if _, ok := param.(string); !ok {
+			return -1, errors.New("param not a string, and cannot be parsed")
+		}
 
-	valueForName, ok := paramsMap[name].(float64)
-	if !ok {
-		return -1, errors.New("missing or otherwise invalid int value for " + name)
-	}
+		if err = json.Unmarshal([]byte(param.(string)), &j); err != nil {
+			return -1, err
+		}
 
-	return int(valueForName), err
+		paramsMap, ok := j.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		valueForName, ok := paramsMap[name].(float64)
+		if !ok {
+			continue
+		}
+
+		return int(valueForName), err
```

Comment:
why does this have to be positive?  If I put in `{"ttl":-500}` I think this would return -500.  Should -500 be an error or should this return an absolute value?

Reply:
After getting background from @DeirdreHolub I've added an error for negative values.